analyses/oncoprint-landscape/03-oncoprint-n-count-table.R

```diff
@@ -101,136 +67,88 @@ option_list <- list(
-          "Lymphoma",
-          "Melanocytic tumor",
-          "Other tumor"
-        )
-      )
-    }
-  # Now filter the remaining data files
-  maf_po_each <- maf_df_po %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  cnv_po_each <- cnv_df_po %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  fusion_po_each <- fusion_df_po %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  # get bs ids in each dataframe
-  maf_po_bsid <- maf_po_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  cnv_po_bsid <- cnv_po_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  fusion_po_bsid <- fusion_po_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  # take union
-  all_po_bsid <- union(maf_po_bsid, cnv_po_bsid) %>% union(fusion_po_bsid) 
-  # write out broad histology and number of samples to a table 
-  oncoprint_n_table_po[i,1] <- histology_of_interest
-  oncoprint_n_table_po[i,2] <- length(all_po_bsid)
-  oncoprint_n_table_po[i,3] <- "primary_only"
-  
-  # Now do the same thing for primary plus samples
-  maf_pp_each <- maf_df_pp %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  cnv_pp_each <- cnv_df_pp %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  fusion_pp_each <- fusion_df_pp %>%
-    dplyr::filter(Tumor_Sample_Barcode %in% metadata_each$Tumor_Sample_Barcode)
-  
-  # get bs ids in each dataframe
-  maf_pp_bsid <- maf_pp_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  cnv_pp_bsid <- cnv_pp_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  fusion_pp_bsid <- fusion_pp_each %>% pull(Tumor_Sample_Barcode) %>% unique()
-  # take union
-  all_pp_bsid <- union(maf_pp_bsid, cnv_pp_bsid) %>% union(fusion_pp_bsid) 
-  # write out broad histology and number of samples to a table 
-  oncoprint_n_table_pp[i,1] <- histology_of_interest
-  oncoprint_n_table_pp[i,2] <- length(all_pp_bsid)
-  oncoprint_n_table_pp[i,3] <- "primary-plus"
-  
-}
-
-# write out n number entering oncoprint
-bind_rows(oncoprint_n_table_pp, oncoprint_n_table_po) %>% 
-  readr::write_tsv(file.path(results_dir, "sample_n_in_oncoprint.tsv"))
+fusion_df <- readr::read_tsv(opt$fusion_file)
+
+#### Set up oncoprint counts ---------------------------------------------------
+
+other_cns_histologies <- c(
+  "Ependymal tumor",
+  "Tumors of sellar region",
+  "Neuronal and mixed neuronal-glial tumor",
+  "Tumor of cranial and paraspinal nerves",
+  "Meningioma",
+  "Mesenchymal non-meningothelial tumor",
+  "Germ cell tumor",
+  "Choroid plexus tumor",
+  "Histiocytic tumor",
+  "Tumor of pineal region",
+  "Metastatic tumors",
+  "Other astrocytic tumor",
+  "Lymphoma",
+  "Melanocytic tumor",
+  "Other tumor"
+)
 
+tumor_barcodes_to_include <- metadata %>%
+  filter(broad_histology %in% c(
+    other_cns_histologies,
+    "Low-grade astrocytic tumor",
```

Comment:
Might be worth coding this other array up earlier when `other_cns_histologies` is defined